src/App.tsx

```diff
@@ -1,8 +1,80 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import { fetchData } from './api/fetchData';
+import { TodosBlock } from './components/TodosBlock';
+import { Todo, User, TodoModified } from './interfaces/data';
 
-const App = () => (
-  <h1>Dynamic list of TODOs</h1>
-);
+const App: React.FC = () => {
+  const API_URL_TODOS = 'https://mate.academy/students-api/todos';
+  const API_URL_USERS = 'https://mate.academy/students-api/users';
+
+  const [todos, setTodos] = useState<TodoModified[]>([]);
+  const [isLoading, setLoading] = useState(false);
+  const [hasError, setError] = useState(false);
+
+  const loadData = () => {
+    let initTodos: Todo[];
+    let users: User[];
+
+    setError(false);
+    setLoading(true);
+
+    fetchData<Todo>(API_URL_TODOS)
+      .then(data => {
+        initTodos = data.data;
+
+        return fetchData<User>(API_URL_USERS);
+      })
+      .then(data => {
+        users = data.data;
```

Comment:
```suggestion
    fetchData<Todo>(API_URL_TODOS)
      .then(data => {
        initTodos = data.data;
      });
      
    fetchData<User>(API_URL_USERS)
        .then(data => {
          users = data.data;
        });
```

сделай что то подобное, очень большая вложенность, тут можно без нее, также у тебя есть async / await

Reply:
Переписал на Promise.all и try-catch